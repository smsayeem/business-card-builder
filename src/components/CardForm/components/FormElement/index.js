import React from 'react'
import PropTypes from 'prop-types'

function FormElement({formField, title, name, type, handleChange}) {
  return (
    <>
      {
        formField === 'input' ? 
        <div className="form-group">
          <label htmlFor={name}>{title}</label>
          <input 
            type={type} 
            name={name}
            className="form-input" 
            onChange={handleChange} />
        </div>
      :
        <div className="form-group">
          <label htmlFor={name}>{title}</label>
            <select name={name} className="form-input" onChange={handleChange} >
              <option value="-1">Select your state</option>
              <option value="NSW">NSW</option>
              <option value="ACT">ACT</option>
              <option value="VIC">VIC</option>
              <option value="QLD">QLD</option>
              <option value="TAS">TAS</option>
              <option value="WA">WA</option>
            </select>
        </div>
      }
    </>
  )
}

FormElement.propTypes = {
  title: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.string,
  handleChange: PropTypes.func,
}

FormElement.defaultProps = {
  title: '',
  name: '',
  type: null,
  handleChange: ()=>{},
}

export default FormElement

