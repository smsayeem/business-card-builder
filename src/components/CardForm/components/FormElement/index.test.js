import { shallow, mount } from 'enzyme';
import React from 'react';
import FormElement from './index';

describe('<FormElement />  Component existence check', () => {
  let wrapper;
  beforeEach(()=>{
    wrapper = shallow(<FormElement />);
  });

  it('should render the <FormElement /> Component correctly', () => {   
    expect(wrapper).toMatchSnapshot();
  });

  // render the <FormElement /> componnet without crashing
  it('should render the <FormElement /> componnet without crashing', ()=>{
    shallow(<wrapper />);
  });

  it('should render lable, input and select tag in the the <FormElement /> Component', () => {
    const inputFormFieldWrapper = shallow(<FormElement formField="input" />)
    expect(inputFormFieldWrapper.find('.form-group').childAt(0).type()).toBe('label');
    expect(inputFormFieldWrapper.find('.form-group').childAt(1).type()).toBe('input');

    const selectFormFieldWrapper = shallow(<FormElement formField="select" />)
    expect(selectFormFieldWrapper.find('.form-group').childAt(0).type()).toBe('label');
    expect(selectFormFieldWrapper.find('.form-group').childAt(1).type()).toBe('select');
  });
});

describe('passing props on <FormElement />)', ()=>{
  const wrapper = mount(<FormElement formField="input" handleChange={()=>{}} name="country" title="country" type="text" />);
  it('accept the CardPreview props', ()=>{
    expect(wrapper.props().formField).toEqual("input");
    expect(wrapper.props().name).toBe('country');
    expect(wrapper.props().title).toEqual('country');
    expect(wrapper.props().type).toEqual('text');
  })
});