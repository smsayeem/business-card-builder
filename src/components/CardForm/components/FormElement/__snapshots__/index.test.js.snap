// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormElement />  Component existence check should render the <FormElement /> Component correctly 1`] = `
<Fragment>
  <div
    className="form-group"
  >
    <label
      htmlFor=""
    />
    <select
      className="form-input"
      name=""
      onChange={[Function]}
    >
      <option
        value="-1"
      >
        Select your state
      </option>
      <option
        value="NSW"
      >
        NSW
      </option>
      <option
        value="ACT"
      >
        ACT
      </option>
      <option
        value="VIC"
      >
        VIC
      </option>
      <option
        value="QLD"
      >
        QLD
      </option>
      <option
        value="TAS"
      >
        TAS
      </option>
      <option
        value="WA"
      >
        WA
      </option>
    </select>
  </div>
</Fragment>
`;
